### DISCUSSION TITLE
Git author must have access to the project on vercel

### DISCUSSION THREAD

**Discussion by julianblanco00** [vercel/community#100] [Help] [answered]
Hi, I can't make deployments in my project. I'm using the same email in my github account, in git and in my vercel member team. The commits are made with the same email that the member of my team but I receive the next message when pushing to github "Git author must have access to the project on vercel"

**Comment by leerob** [comment, marked as answer]
Could you provide a deployment ID, commit SHA and/or your account slug? Thanks.

**Reply by mcpat1993** [reply]
Same issue on commit 322149684c5a7a6a017bf4ca713c3fd769f073ca

**Reply by Pieparker** [reply]
@mcpat1993 ~It looks like the project in question, `prj_AXXyHusoCiNePv2S6eJDN0cXzGSC`, is not connected to a repository via Git~. You'll want to ensure that you [check out this article](https://vercel.com/support/articles/why-aren-t-commits-triggering-deployments-on-vercel) for more useful tidbits!

Should issues persist, for this teams-specific issue, we'd encourage you to reach out via the [support form](https://vercel.com/support).

**Reply by mcpat1993** [reply]
I've actually since gotten the following:
<img width="694" alt="Screen Shot 2022-02-01 at 14 41 48" src="https://user-images.githubusercontent.com/8905548/151989322-bcf2a293-09c8-464a-a650-00b7ecefeed0.png">

So now after waiting a few days deployment has completed. I haven't changed anything since, I just tried it again because someone in my team told me that it has happened before and that new contributors need a few days to sync permissions with Vercel. 

It should show a connection with a Git repository as well as that is where I've made the commit to and which has automatically triggered Vercel deployment.

It would be helpful to me to know why this 'few days' of wait is necessary/what is going on behind the scenes. Appreciate any guidance.

**Reply by Pieparker** [reply]
I suspect that this is something erroneous but transient between GitHub and our integration which is usually resolved with the [above advice](https://vercel.com/support/articles/why-aren-t-commits-triggering-deployments-on-vercel). However, I'll ask our builds team if there's any more rhyme or reason to this behaviour.

internal ref: https://github.com/vercel/customer-issues/issues/249

**Reply by Pieparker** [reply]
Hey there! Wanted to let you know that we've not forgotten about this. It's proving to be an interesting one for our team! We're looking more in to the trickle-down permissions between GitHub and our platform.

Is there anything else you can share about your GitHub team or organisation; do you make use of SAML SSO or are there any other advanced GitHub features that your org uses?

**Reply by mcpat1993** [reply]
Hey! Thanks or staying on it! It's only a minor inconvenience as the trickling did eventually happen after a week inexplicably but it would be nice for those permissions to be instantaneous.

We do not use SSO. Could you be more specific about what you mean by 'advanced GitHub features' and I'd be happy to let you know if we're using them.

**Reply by Pieparker** [reply]
Understood! Undesirable behaviour is exactly that; we'll keep digging!

Thanks for that. To be honest, that was the biggie! There's nothing else at this stage that I or the team were conscious of. We'll loop back if there's anything in particular we don't think is playing nicely & would like to find out 👀.

**Reply by Pieparker** [reply]
Thanks for your patience! The team took a deep dive in to the timeline of events here:

- We received the commit `322149684c5a7a6a017bf4ca713c3fd769f073ca` on Jan 24 23:09:53.989
- Our system then tries to find a Vercel user, and found you; all good so far
- We are able to observe that you were _invited_ to the team before this time
- We are able to observe that you _joined_ the team after this time

It appears as though you had been invited but hadn't joined the team/accepted the invitation when the commit was received.

The only time we will then re-run a commit is when you follow the "request access" flow, and not the email invitation flow, because we only know the exact commit related to a team joiner when using the "request access" flow.

Thank you again for bearing with us and you know where we are, if you have any follow-ups!

**Reply by grinsteindavid** [reply]
I do have same thing with my team

**Reply by abdulsamadayoade** [reply]
I'm facing the same issue too

**Reply by amyegan** [reply]
Is the problem happening with every new commit, or only occurring for commits that happened before joining the team?

**Reply by RahulGuptaIIITA** [reply]
I'm facing the similar issue. My email id has a github account, the same email id is added to vercel teams, and to the repository which i'm trying to deploy.

**Reply by amyegan** [reply]
If you can share a deployment ID, commit SHA and/or your account slug, then we can look into the logs to get a better idea of what is going wrong for you 🙏

**Reply by abdulsamadayoade** [reply]
> Is the problem happening with every new commit, or only occurring for commits that happened before joining the team?

seems to be working fine now, didn't do anything different

**Reply by LouGel** [reply]
I have the same issue , i added myself and vercel tell me that i'm not a github user, nor a vercel team member

**Reply by LouGel** [reply]
The only way it worked it's by deleteting and relink the project

**Comment by williamaditama** [comment]
I have the same issue

**Reply by amyegan** [reply]
Let's continue the discussion in https://github.com/vercel/community/discussions/1126
